app/controllers/user_posts_controller.rb

```diff
@@ -0,0 +1,10 @@
+class UserPostsController < ApplicationController
+  def index
+    @user_id = params[:user_id]
+  end
+
+  def show
+    @user_id = params[:user_id]
+    @id = params[:id]
+  end
+end
```

Comment:
- [ ] I suggest that you rename this controller from `UserPostsController` to `PostsController`